### DISCUSSION TITLE
Cache flush required every time after changing content.

### DISCUSSION THREAD

**Discussion by kolaente** [statamic/cms#5906] [Troubleshooting]
I'm not sure this is intended behaviour or just a configuration problem.

Every time after changing content, I have to do a full cache flush to make the changes visible on the site. The site is using redis in prod, but this also happens when developing locally without redis.

**Comment by jasonvarga** [comment]
Are you changing content in the control panel or directly in files?

If you're editing files directly, make sure you have the Stache watcher enabled. It's enabled by default, but we recommend disabling it on production. Maybe you disabled it.

https://statamic.dev/stache#watcher

**Reply by kolaente** [reply]
I have the Stache watcher enabled (looks like it was enabled on Prod as well). It does not seem to make a difference though, when I edit pages locally the cache is only invalidated when I edit through the control panel. On Prod it is only invalidated when I clear it manually, even when editing files through the control panel.

This also happens with antlers template files during development. (locally, `APP_ENV=local`)

**Reply by jasonvarga** [reply]
Are you using static caching? That would not be invalided when you edit a file.

**Reply by edalzell** [reply]
It might be this issue: https://github.com/statamic/cms/issues/3291

We've seen some cache invalidation weirdness crop up recently as well, haven't figured out how to reliably reproduce it.

Are you able to set your cache to `file` and see if that resolved it @kolaente?

**Reply by kolaente** [reply]
If I understood it correctly, I am using static caching via Laravel's configured cache driver. Here is my config:

```php
<?php

return [

    'strategy' => 'half',

    'strategies' => [

        'half' => [
            'driver' => 'application',
            'expiry' => null,
        ],

        'full' => [
            'driver' => 'file',
            'path' => public_path('static'),
            'lock_hold_length' => 0,
        ],

    ],

    'exclude' => [
        //
    ],

    'invalidation' => [

        'class' => null,

        'rules' => [
            'collections' => [
                'news' => [
                    'urls' => [
                        '/news',
                    ],
                ],
            ],
        ],

    ],

];

```

But I thought using the half measure would invalidate the cached files whenever I save an item through the control panel?

@edalzell You mean configuring Laravel's cache driver to file (instead of redis on prod) or using the full cache measure?
